In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np

# File to Load (Remember to Change These)
file_to_load ="Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

## Player Count

* Display the total number of players


In [2]:
#Use length of SN for number or players
number_of_players=len(purchase_data["SN"].value_counts())

#create data frame
player_count=pd.DataFrame({'Total Players':[number_of_players]})
player_count

Total Players
0            576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [3]:
price_df=purchase_data[["Item Name","Price"]]

#Number of Unique Items
unique_items=len((purchase_data["Item Name"]).unique())

#Average Purchase Price
average_price=round(purchase_data["Price"].mean(),2)

#Total Number of Purchases
number_of_purchase=purchase_data["Price"].count()

#Total Revenue
purchase_total=purchase_data["Price"].sum()

#Create Data Frame
purchasing_analysis=pd.DataFrame({'Number of Unique Items':[unique_items],'Average Price':[average_price],
                                  'Number of Purchases':[number_of_purchase],'Total Revenue':[purchase_total]})
purchasing_analysis.style.format({'Average Price':"${:,.2f}",'Total Revenue':'${:,.2f}'})



## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [4]:
#Group purchase by gender
gender_info=purchase_data.groupby("Gender")

#Count of genders (male and female)
gender_count = gender_info.nunique()["SN"]

#Percentage of genders (male and female)
gender_percentage = round(gender_count/number_of_players * 100,2)

#Create data frame
gender_demo=pd.DataFrame({'Total Count':gender_count,'Percentage of Players %': gender_percentage})
gender_demo.index.name=None
gender_demo.sort_values(["Total Count"], ascending = False).style.format({"Percentage of Players %":"{:.2f}%"})



## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [5]:
#Group purchase by gender
gender_info=purchase_data.groupby("Gender")

#Count screen names
total_count_gender = gender_info.nunique()["SN"]

#Purchase Count
number_of_purchase=gender_info["Purchase ID"].count()

#Average Purchase Price
average_price=round(gender_info["Price"].mean(),2)

#Average purchase total by gender 
avg_purchase_total =gender_info["Price"].sum()

#Average purchase total per person
avg_purchase_person=avg_purchase_total/total_count_gender

#Create dataframe
gender_demo=pd.DataFrame({"Purchase Count":number_of_purchase,"Average Purchase Price":average_price,"Total Purchase Value":avg_purchase_total,"Avg Total Purchase per Person":avg_purchase_person})
gender_demo.style.format({'Average Purchase Price':"${:,.2f}",'Total Purchase Value':'${:,.2f}','Avg Total Purchase per Person':'${:,.2f}'})



## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [6]:
#Establish bins for ages
bin_for_ages = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 99999]
Age_group_label = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

#Categorize the existing players using the age bins. Hint: use pd.cut()
purchase_data["Age Group"] = pd.cut(purchase_data["Age"],bin_for_ages,labels=Age_group_label)
purchase_data

#Calculate the numbers and percentages by age group
age_group=purchase_data.groupby('Age Group')
total_count_by_age=age_group["SN"].nunique()
percentage_by_age=round((total_count_by_age/number_of_players)*100,2)

#create data frame
age_demo=pd.DataFrame({"Total Count":total_count_by_age,"Percentage of Players %":percentage_by_age})
age_demo.style.format({"Percentage of Players %":"{:,.2f}%"})

 

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [7]:
#Bin the purchase_data data frame by age

#Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
purchase_age_count=age_group['Purchase ID'].count()
avg_purchase_price=round(age_group["Price"].mean(),2)
total_purchase_value=age_group['Price'].sum()
total_per_person=round(total_purchase_value/total_count_by_age,2)

#Create a summary data frame to hold the results
#Create data frame
purchase_age=pd.DataFrame({"Purchase Count":purchase_age_count,"Average Purchase Price":avg_purchase_price,"Total Purchase Value":total_purchase_value,"Avg Total Purchase per Person":total_per_person})      
purchase_age.style.format({"Average Purchase Price":"${:,.2f}","Total Purchase Value":"${:,.2f}","Avg Total Purchase per Person":"${:,.2f}"})

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [8]:
#Run basic calculations to obtain the results in the table below
spender_info=purchase_data.groupby("SN")
count_spenders=spender_info["Purchase ID"].count()
avg_purchase_spenders=round(spender_info["Price"].mean(),2)
total_purchase_spenders=spender_info['Price'].sum()

#Create a summary data frame to hold the results
top_spenders = pd.DataFrame({"Purchase Count": count_spenders,"Average Purchase Price": avg_purchase_spenders,"Total Purchase Value":total_purchase_spenders})

#Sort the total purchase value column in descending order
top_spender = top_spenders.sort_values(["Total Purchase Value"], ascending=False).head()


top_spender.style.format({"Average Purchase Price":"${:,.2f}","Total Purchase Value":"${:,.2f}"})

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, average item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [9]:
#Retrieve the Item ID, Item Name, and Item Price columns
item_data = purchase_data[["Item ID", "Item Name", "Price"]]
item_info = item_data.groupby(["Item ID","Item Name"])
                          
#Group by Item ID and Item Name. Perform calculations to obtain purchase count, average item price, and total purchase value
purchase_count_item = item_info["Price"].count()
total_purchase_value = (item_info["Price"].sum()) 
average_item_price = total_purchase_value/purchase_count_item

# Create data frame with obtained values
most_popular_items = pd.DataFrame({"Purchase Count": purchase_count_item,"Item Price": average_item_price,"Total Purchase Value":total_purchase_value})

# Sort in descending order to obtain top spender names and provide top 5 item names
format_most_popular_items = most_popular_items.sort_values(["Purchase Count"], ascending=False).head()

# Format with currency style
format_most_popular_items.style.format({"Item Price":"${:,.2f}","Total Purchase Value":"${:,.2f}"})

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [10]:
#Sort the above table by total purchase value in descending order
format_most_popular_items= most_popular_items.sort_values(["Total Purchase Value"],ascending=False).head()
#Optional: give the displayed data cleaner formatting
format_most_popular_items.style.format({"Item Price":"${:,.2f}","Total Purchase Value":"${:,.2f}"})
#Display a preview of the data frame

## Data Analysis and Conclusions:


1. Out of the 576 total players the gender demographics percentage Male accounts for, 84.03% followed by Female, 14.06%. Male also topped the list on most purchases at 1,967.64 the female gender only for 361.94.  It clearly shows video gaming is a male dominated market and the numbers from female gamers are slim compare to male, we could make the assumption it's not being marketed well to women or they have less downtime than the male audience.


2. On the age demographics, the 20-24 age group has the highest purchases at 44.79%, followed by the 15-19 age group at 18.58% and thirdly by the 25-29 age group at 13.37%. Video gaming trend up to the youger generation and the percentage trends down to the much older age group bracket, 30-34 at 9.03%, 35-39 at 5.38% and 40+ at 2.08%.

3. The top three spenders are Lisosia93, Idastidru52, and Chamjask73.  In terms of most popular the top three are "Final Critic", "Oathbreaker, Last Hope of the Breaking Storm" and "Fiery Glass Crusader".  For the most profitable items it's pretty similar with "Final Critic", "Oathbreaker, Last Hope of the Breaking Storm" and "Nirvana" making up the list. Nirvana slightly edge Fiery Glass Crusader and dislodged from 3rd place. 
